ariadne/src/Ariadne/Cardano/Face.hs

```diff
@@ -56,9 +98,149 @@ data CardanoEvent
+  , MFunctor' d CardanoModeMonad Production
+  )
+  => Mockable d CardanoMode where
+  liftMockable = CardanoMode . liftMockable . hoist' unwrapCardanoMode
+
+-- ♫ E-E-G-G-E-E-G Mr. Boilerplate!
+-- Amazing, with Mr. Boilerplate my code is now twice as long!!!
+
+instance HasNodeType CardanoContext where
+    getNodeType = getNodeType . ccRealModeContext
+
+instance {-# OVERLAPPABLE #-}
+    HasLens tag (RealModeContext EmptyMempoolExt) r =>
+    HasLens tag CardanoContext r
+  where
+    lensOf = ccRealModeContextL . lensOf @tag
+
+instance HasSscContext CardanoContext where
+    sscContext = ccRealModeContextL . sscContext
+
+instance HasPrimaryKey CardanoContext where
+    primaryKey = ccRealModeContextL . primaryKey
+
+instance HasMisbehaviorMetrics CardanoContext where
+    misbehaviorMetrics = ccRealModeContextL . misbehaviorMetrics
+
+instance HasUserSecret CardanoContext where
+    userSecret = ccRealModeContextL . userSecret
+
+instance HasShutdownContext CardanoContext where
+    shutdownContext = ccRealModeContextL . shutdownContext
+
+instance HasSlottingVar CardanoContext where
+    slottingTimestamp = ccRealModeContextL . slottingTimestamp
+    slottingVar = ccRealModeContextL . slottingVar
+
+instance HasSlogContext CardanoContext where
+    slogContext = ccRealModeContextL . slogContext
+
+instance HasSlogGState CardanoContext where
+    slogGState = ccRealModeContextL . slogGState
+
+instance HasNodeContext CardanoContext where
+    nodeContext = ccRealModeContextL . nodeContext
+
+instance HasLoggerName' CardanoContext where
+    loggerName = ccRealModeContextL . loggerName
+
+instance HasJsonLogConfig CardanoContext where
+    jsonLogConfig = ccRealModeContextL . jsonLogConfig
+
+instance {-# OVERLAPPING #-} CanJsonLog CardanoMode where
+    jsonLog = jsonLogDefault
+
+instance (HasConfiguration, MonadSlotsData ctx CardanoMode)
+      => MonadSlots ctx CardanoMode
+  where
+    getCurrentSlot = getCurrentSlotSimple
+    getCurrentSlotBlocking = getCurrentSlotBlockingSimple
+    getCurrentSlotInaccurate = getCurrentSlotInaccurateSimple
+    currentTimeSlotting = currentTimeSlottingSimple
+
+instance HasConfiguration => MonadGState CardanoMode where
+    gsAdoptedBVData = gsAdoptedBVDataDefault
+
+instance HasConfiguration => MonadDBRead CardanoMode where
+    dbGet = dbGetDefault
+    dbIterSource = dbIterSourceDefault
+    dbGetSerBlock = dbGetSerBlockRealDefault
+    dbGetSerUndo = dbGetSerUndoRealDefault
+
+instance HasConfiguration => MonadDB CardanoMode where
+    dbPut = dbPutDefault
+    dbWriteBatch = dbWriteBatchDefault
+    dbDelete = dbDeleteDefault
+    dbPutSerBlunds = dbPutSerBlundsRealDefault
+
+instance (HasConfiguration, HasTxpConfiguration) =>
+         MonadTxpLocal CardanoMode where
+    txpNormalize = txNormalize
+    txpProcessTx = txProcessTransaction
+
+instance MonadReporting CardanoMode where
+    report rt = Mtl.ask >>= liftIO . flip runReporter rt . (rmcReporter . ccRealModeContext)
```

Comment:
I think all these instances should reuse implementations for `RealMode`. In this version definitions are mostly copy-pasted from those in `Pos.WorkMode`.
See how instances are defined [here](https://github.com/input-output-hk/cardano-sl/blob/226186092edd826c469850d2d808df9720f08efc/auxx/src/Mode.hs) (using `realModeToAuxx` mostly).

Reply:
This instance is still copy-pasted :unamused:

Reply:
Please fix.